### PR TITLE
LTD-1683-1667-en-gb-pdf-and-title

### PR DESCRIPTION
**Pull request by seijihg**
### Aim

Updated weasyprint since the older version did not support lang="en-gb".

SS:

Screenshot 2023-03-25 at 08 23 35
Screenshot 2023-03-27 at 21 33 55

[LTD-1683](https://uktrade.atlassian.net/browse/LTD-1683)
[LTD-1667](https://uktrade.atlassian.net/browse/LTD-1667)

[LTD-1683]: https://uktrade.atlassian.net/browse/LTD-1683?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LTD-1667]: https://uktrade.atlassian.net/browse/LTD-1667?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kevincarrogan — COMMENTED** [review]
Could you add some tests please.

File: api/letter_templates/templates/letter_templates/application_form.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {% load custom_tags %}
 
 {{ css|safe }}
```
Comment (kevincarrogan) [inline api/letter_templates/templates/letter_templates/application_form.html:3]: The CSS being outside the `html` might cause problems at some point.

File: api/letter_templates/templates/letter_templates/application_form.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
-
+<html lang="en-gb">
```
Comment (kevincarrogan) [inline api/letter_templates/templates/letter_templates/application_form.html:4]: I think a `html` block without a `body` is technically invalid.

File: api/letter_templates/templates/letter_templates/siel.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {% load static custom_tags %}
 
 {{ css|safe }}
-
+<html lang="en-gb">
```
Comment (kevincarrogan) [inline api/letter_templates/templates/letter_templates/siel.html:4]: These all seem to start with the same thing. Could we extract this out into a base template?

Comment (seijihg) [reply api/letter_templates/templates/letter_templates/siel.html]: yes and done :D

File: api/letter_templates/templates/letter_templates/application_form.html (hunk 1 above)
Comment (seijihg) [reply api/letter_templates/templates/letter_templates/application_form.html]: we are just generating for the PDF so I don't think it matters much.

File: api/cases/generated_documents/tests/test_generate_document.py [hunk 4]
```diff
@@ -390,3 +393,28 @@ def test_get_generated_documents_not_visible_to_exporter_exporter_user_success(s
...
+            "siel": TemplateTitles.SIEL,
+        }
+
+        for layout in layouts:
+            html = generate_preview(layout=layout, case=case, text="")
+            pdf = html_to_pdf(html, layout, None)
+            with io.BytesIO(pdf) as open_pdf_file:
+                reader = PdfFileReader(open_pdf_file)
+                meta = reader.getDocumentInfo()
+                self.assertTrue(meta.title, f"{titles[layout]} {case.reference_code}")
```
Comment (kevincarrogan) [inline api/cases/generated_documents/tests/test_generate_document.py:420]: This should be `self.assertEqual`. At the moment you're just checking that `meta.title` exists and not actually checking that it's the correct value.

File: api/cases/generated_documents/tests/test_generate_document.py [hunk 5]
```diff
@@ -390,3 +393,28 @@ def test_get_generated_documents_not_visible_to_exporter_exporter_user_success(s
...
+            "siel",
+        ]
+        titles = {
+            "application_form": TemplateTitles.APPLICATION_FORM,
+            "nlr": TemplateTitles.NLR,
+            "refusal": TemplateTitles.REFUSAL_LETTER,
+            "siel": TemplateTitles.SIEL,
+        }
+
+        for layout in layouts:
```
Comment (kevincarrogan) [inline api/cases/generated_documents/tests/test_generate_document.py:414]: Could we use parameterized here?

File: api/cases/generated_documents/tests/test_generate_document.py (hunk 4 above)
Comment (kevincarrogan) [inline api/cases/generated_documents/tests/test_generate_document.py:420]: Could we get a similar check for language?

File: api/letter_templates/templates/letter_templates/base.html [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<html lang="en-gb">
+
+<meta charset="utf-8">
+<title>{{title}} {{ case_reference }}</title>
+{{ css|safe }}
```
[the remark this answers is not in the document]
Comment (seijihg) [reply api/letter_templates/templates/letter_templates/base.html:5]: Hi, as previously I don’t think you need end tags for them. I run them in W3 Validator and its fine. You dont actually require them unless xhtml. And again is for PDF so its fine. The preview page still looks exactly the same as before I did the changes.

**Review by kevincarrogan — APPROVED** [review]

**Review by depsiatwal — APPROVED** [review]
Looks good to me

File: api/letter_templates/helpers.py [hunk 7]
```diff
@@ -57,8 +57,18 @@ def generate_preview(
     include_css=True,
 ):
     template_name = f"letter_templates/{layout}.html"
-
-    context = {"include_digital_signature": include_digital_signature, "user_content": text}
+    title = ""
```
Comment (currycoder) [inline api/letter_templates/helpers.py:60]: Out of interest, is there another function which generates more than just a preview?

**Review by currycoder — APPROVED** [review]

File: api/letter_templates/helpers.py (hunk 7 above)
Comment (seijihg) [reply api/letter_templates/helpers.py:60]: Not something I am aware of ... :D

**Pull request merged**